src/config.rs

```diff
@@ -59,6 +59,9 @@ pub struct Config {
     pub run_only_ignored: bool,
     /// Filters must match exactly instead of just checking for substrings.
     pub filter_exact: bool,
+    /// Prefix added to all diagnostic code matchers. Note this will make it impossible
+    /// match codes which do not contain this prefix.
+    pub diagnostic_code_prefix: Option<String>,
```

Comment:
if we are adding a config option anyway, we may as well go all the way and have a mode for `//~ LEVEL` and one for `//~ CODE`, and only permit the one selected.

Reply:
I believe we a few tests that have rustc diagnostics as well as clippy ones to check for overlaps, so they would need both kinds

Reply:
We might also have a few lints which would benefit from having the messages checked as well.